core/block/editor/smartblock/smartblock.go

```diff
@@ -699,7 +699,7 @@ func (sb *smartBlock) Apply(s *state.State, flags ...ApplyFlag) (err error) {
 				sb.undo.Add(act)
 			}
 		}
-	} else if hasStoreChanges(changes) || migrationVersionUpdated { // TODO: change to len(changes) > 0
+	} else if hasStoreChanges(changes) || migrationVersionUpdated || hasNotificationChanges(changes) { // TODO: change to len(changes) > 0
```

Comment:
Do we need two functions here btw? Or they are used only here? If that is the case we can have only function hasChanges